[BUG] Repeated warnings during compile about `warning: "DEBUG_ECHOPGM_P" redefined`

<!--

Please follow the instructions below. Failure to do so may result in your issue being closed.

### Before Reporting a Bug

1. Test with the `bugfix-2.0.x` branch to see whether the issue still exists.

2. Get troubleshooting help from the Marlin community to confirm it's a bug and not just a configuration error. Links at https://github.com/MarlinFirmware/Marlin/issues/new/choose

### Instructions

1. Fill out every section of the template below.

2. Always attach configuration files, regardless of whether you think they are involved.

3. Read and understand Marlin's Code of Conduct. By filing an Issue, you are expected to comply with it, including treating everyone with respect: https://github.com/MarlinFirmware/Marlin/blob/master/.github/code_of_conduct.md

-->

### Bug Description

Sometime in the last week I've started getting a lot of warnings during compile about `warning: "DEBUG_ECHOPGM_P" redefined`, possibly related to #21231. It still compiles fine, but figured since it shows up hundreds of times I'd mention it.

### Configuration Files

**Required:** Include a ZIP file containing `Configuration.h` and `Configuration_adv.h`.
[Marlin SKR 1.4 Turbo configs 2021-03-06.zip](https://github.com/MarlinFirmware/Marlin/files/6095977/Marlin.SKR.1.4.Turbo.configs.2021-03-06.zip)

If you've made any other modifications describe them in detail here.

### Steps to Reproduce

<!-- Describe the steps needed to reproduce the issue. (You can remove this invisible comment.) -->

1. [First Step]
2. [Second Step]
3. [and so on...]

**Expected behavior:**

Marlin should compile without a huge amount of warnings

**Actual behavior:**

Lots of repeated warnings

#### Additional Information

Complete log file with compile warnings: [https://pastebin.com/4f3WVaE6](https://pastebin.com/4f3WVaE6)

```
Compiling .pio\build\LPC1769\src\src\core\utility.cpp.o
Compiling .pio\build\LPC1769\src\src\feature\babystep.cpp.o
Compiling .pio\build\LPC1769\src\src\feature\bedlevel\bedlevel.cpp.o
In file included from Marlin\src\feature/bedlevel/ubl/ubl.h:328,
                 from Marlin\src\feature/bedlevel/bedlevel.h:67,
                 from Marlin\src\MarlinCore.cpp:175:
Marlin\src\feature/bedlevel/ubl/../../../core/debug_out.h:92: warning: "DEBUG_ECHOPGM_P" redefined
   92 |   #define DEBUG_ECHOPGM_P(P)          NOOP
      |
In file included from Marlin\src\feature/bedlevel/ubl/ubl.h:29,
                 from Marlin\src\feature/bedlevel/bedlevel.h:67,
                 from Marlin\src\MarlinCore.cpp:175:
Compiling .pio\build\LPC1769\src\src\feature\bedlevel\ubl\ubl.cpp.o
Compiling .pio\build\LPC1769\src\src\feature\bedlevel\ubl\ubl_G29.cpp.o
Marlin\src\feature/bedlevel/ubl/../../../core/debug_out.h:62: note: this is the location of the previous definition
Compiling .pio\build\LPC1769\src\src\feature\bedlevel\ubl\ubl_motion.cpp.o
   62 |   #define DEBUG_ECHOPGM_P(P)      SERIAL_ECHOPGM_P(P)
Compiling .pio\build\LPC1769\src\src\feature\fwretract.cpp.o
Compiling .pio\build\LPC1769\src\src\feature\leds\leds.cpp.o
      | 
Compiling .pio\build\LPC1769\src\src\feature\leds\neopixel.cpp.o
Compiling .pio\build\LPC1769\src\src\feature\leds\printer_event_leds.cpp.o
In file included from Marlin\src\core\../feature/bedlevel/ubl/ubl.h:328,
Compiling .pio\build\LPC1769\src\src\feature\pause.cpp.o
                 from Marlin\src\core\../feature/bedlevel/bedlevel.h:67,
                 from Marlin\src\core\utility.cpp:56:
Marlin\src\core\../feature/bedlevel/ubl/../../../core/debug_out.h:92: warning: "DEBUG_ECHOPGM_P" redefined
   92 |   #define DEBUG_ECHOPGM_P(P)          NOOP
      | 
In file included from Marlin\src\core\../feature/bedlevel/ubl/ubl.h:29,
                 from Marlin\src\core\../feature/bedlevel/bedlevel.h:67,
                 from Marlin\src\core\utility.cpp:56:
Compiling .pio\build\LPC1769\src\src\feature\runout.cpp.o
Marlin\src\core\../feature/bedlevel/ubl/../../../core/debug_out.h:62: note: this is the location of the previous definition
   62 |   #define DEBUG_ECHOPGM_P(P)      SERIAL_ECHOPGM_P(P)
      | 
Compiling .pio\build\LPC1769\src\src\feature\tmc_util.cpp.o
Compiling .pio\build\LPC1769\src\src\gcode\bedlevel\G42.cpp.o
In file included from Marlin\src\feature\bedlevel\ubl/ubl.h:328,
Compiling .pio\build\LPC1769\src\src\gcode\bedlevel\M420.cpp.o
                 from Marlin\src\feature\bedlevel\bedlevel.h:67,
Compiling .pio\build\LPC1769\src\src\gcode\bedlevel\ubl\G29.cpp.o
                 from Marlin\src\feature\bedlevel\bedlevel.cpp:27:
Marlin\src\feature\bedlevel\ubl/../../../core/debug_out.h:92: warning: "DEBUG_ECHOPGM_P" redefined
Compiling .pio\build\LPC1769\src\src\gcode\bedlevel\ubl\M421.cpp.o
   92 |   #define DEBUG_ECHOPGM_P(P)          NOOP
      | 
In file included from Marlin\src\feature\bedlevel\ubl/ubl.h:29,
Compiling .pio\build\LPC1769\src\src\gcode\calibrate\G28.cpp.o
                 from Marlin\src\feature\bedlevel\bedlevel.h:67,
                 from Marlin\src\feature\bedlevel\bedlevel.cpp:27:
Compiling .pio\build\LPC1769\src\src\gcode\calibrate\G33.cpp.o
Marlin\src\feature\bedlevel\ubl/../../../core/debug_out.h:62: note: this is the location of the previous definition
   62 |   #define DEBUG_ECHOPGM_P(P)      SERIAL_ECHOPGM_P(P)
      |
In file included from Marlin\src\feature\bedlevel\bedlevel.cpp:43:
Marlin\src\feature\bedlevel\../../core/debug_out.h:62: warning: "DEBUG_ECHOPGM_P" redefined
   62 |   #define DEBUG_ECHOPGM_P(P)      SERIAL_ECHOPGM_P(P)
      | 
In file included from Marlin\src\feature\bedlevel\ubl/ubl.h:328,
                 from Marlin\src\feature\bedlevel\bedlevel.h:67,
                 from Marlin\src\feature\bedlevel\bedlevel.cpp:27:
Marlin\src\feature\bedlevel\ubl/../../../core/debug_out.h:92: note: this is the location of the previous definition
   92 |   #define DEBUG_ECHOPGM_P(P)          NOOP
      |
In file included from Marlin\src\feature\bedlevel\ubl\../ubl/ubl.h:328,
                 from Marlin\src\feature\bedlevel\ubl\../bedlevel.h:67,
Compiling .pio\build\LPC1769\src\src\gcode\calibrate\M48.cpp.o
                 from Marlin\src\feature\bedlevel\ubl\ubl_G29.cpp:27:
Marlin\src\feature\bedlevel\ubl\../ubl/../../../core/debug_out.h:92: warning: "DEBUG_ECHOPGM_P" redefined
   92 |   #define DEBUG_ECHOPGM_P(P)          NOOP
Compiling .pio\build\LPC1769\src\src\gcode\calibrate\M665.cpp.o
      | 
In file included from Marlin\src\feature\bedlevel\ubl\../ubl/ubl.h:29,
                 from Marlin\src\feature\bedlevel\ubl\../bedlevel.h:67,
                 from Marlin\src\feature\bedlevel\ubl\ubl_G29.cpp:27:
Compiling .pio\build\LPC1769\src\src\gcode\calibrate\M666.cpp.o
Marlin\src\feature\bedlevel\ubl\../ubl/../../../core/debug_out.h:62: note: this is the location of the previous definition
   62 |   #define DEBUG_ECHOPGM_P(P)      SERIAL_ECHOPGM_P(P)
      |
In file included from Marlin\src\feature\bedlevel\ubl\../ubl/ubl.h:328,
Compiling .pio\build\LPC1769\src\src\gcode\config\M200-M205.cpp.o
                 from Marlin\src\feature\bedlevel\ubl\../bedlevel.h:67,
                 from Marlin\src\feature\bedlevel\ubl\ubl_motion.cpp:26:
Marlin\src\feature\bedlevel\ubl\../ubl/../../../core/debug_out.h:92: warning: "DEBUG_ECHOPGM_P" redefined
   92 |   #define DEBUG_ECHOPGM_P(P)          NOOP
Compiling .pio\build\LPC1769\src\src\gcode\config\M220.cpp.o
      | 
In file included from Marlin\src\feature\bedlevel\ubl\../ubl/ubl.h:29,
                 from Marlin\src\feature\bedlevel\ubl\../bedlevel.h:67,
                 from Marlin\src\feature\bedlevel\ubl\ubl_motion.cpp:26:
Marlin\src\feature\bedlevel\ubl\../ubl/../../../core/debug_out.h:62: note: this is the location of the previous definition
   62 |   #define DEBUG_ECHOPGM_P(P)      SERIAL_ECHOPGM_P(P)
      |
In file included from Marlin\src\feature\bedlevel\ubl\ubl_G29.cpp:46:
Compiling .pio\build\LPC1769\src\src\gcode\config\M221.cpp.o
Marlin\src\feature\bedlevel\ubl\../../../core/debug_out.h:62: warning: "DEBUG_ECHOPGM_P" redefined
   62 |   #define DEBUG_ECHOPGM_P(P)      SERIAL_ECHOPGM_P(P)
Compiling .pio\build\LPC1769\src\src\gcode\config\M301.cpp.o
      | 
In file included from Marlin\src\feature\bedlevel\ubl\../ubl/ubl.h:328,
                 from Marlin\src\feature\bedlevel\ubl\../bedlevel.h:67,
                 from Marlin\src\feature\bedlevel\ubl\ubl_G29.cpp:27:
Marlin\src\feature\bedlevel\ubl\../ubl/../../../core/debug_out.h:92: note: this is the location of the previous definition
   92 |   #define DEBUG_ECHOPGM_P(P)          NOOP
      |
In file included from Marlin\src\feature\bedlevel\ubl\../ubl/ubl.h:328,
                 from Marlin\src\feature\bedlevel\ubl\../bedlevel.h:67,
                 from Marlin\src\feature\bedlevel\ubl\ubl.cpp:27:
Marlin\src\feature\bedlevel\ubl\../ubl/../../../core/debug_out.h:92: warning: "DEBUG_ECHOPGM_P" redefined
   92 |   #define DEBUG_ECHOPGM_P(P)          NOOP
      |
In file included from Marlin\src\feature\bedlevel\ubl\../ubl/ubl.h:29,
                 from Marlin\src\feature\bedlevel\ubl\../bedlevel.h:67,
                 from Marlin\src\feature\bedlevel\ubl\ubl.cpp:27:
Marlin\src\feature\bedlevel\ubl\../ubl/../../../core/debug_out.h:62: note: this is the location of the previous definition
   62 |   #define DEBUG_ECHOPGM_P(P)      SERIAL_ECHOPGM_P(P)
      |
In file included from Marlin\src\gcode\bedlevel\../../feature/bedlevel/ubl/ubl.h:328,
                 from Marlin\src\gcode\bedlevel\../../feature/bedlevel/bedlevel.h:67,
                 from Marlin\src\gcode\bedlevel\G42.cpp:31:
Marlin\src\gcode\bedlevel\../../feature/bedlevel/ubl/../../../core/debug_out.h:92: warning: "DEBUG_ECHOPGM_P" redefined
   92 |   #define DEBUG_ECHOPGM_P(P)          NOOP
      | 
In file included from Marlin\src\gcode\bedlevel\../../feature/bedlevel/ubl/ubl.h:29,
                 from Marlin\src\gcode\bedlevel\../../feature/bedlevel/bedlevel.h:67,
                 from Marlin\src\gcode\bedlevel\G42.cpp:31:
Marlin\src\gcode\bedlevel\../../feature/bedlevel/ubl/../../../core/debug_out.h:62: note: this is the location of the previous definition
   62 |   #define DEBUG_ECHOPGM_P(P)      SERIAL_ECHOPGM_P(P)
      |
In file included from Marlin\src\gcode\bedlevel\../../feature/bedlevel/ubl/ubl.h:328,
                 from Marlin\src\gcode\bedlevel\../../feature/bedlevel/bedlevel.h:67,
Compiling .pio\build\LPC1769\src\src\gcode\config\M302.cpp.o
                 from Marlin\src\gcode\bedlevel\M420.cpp:28:
Marlin\src\gcode\bedlevel\../../feature/bedlevel/ubl/../../../core/debug_out.h:92: warning: "DEBUG_ECHOPGM_P" redefined
   92 |   #define DEBUG_ECHOPGM_P(P)          NOOP
      |
In file included from Marlin\src\gcode\bedlevel\ubl\../../../feature/bedlevel/ubl/ubl.h:328,
Compiling .pio\build\LPC1769\src\src\gcode\config\M304.cpp.o
                 from Marlin\src\gcode\bedlevel\ubl\../../../feature/bedlevel/bedlevel.h:67,
                 from Marlin\src\gcode\bedlevel\ubl\M421.cpp:32:
c:\users\light\onedrive\documents\github\marlin_q5\marlin\src\core\debug_out.h:92: warning: "DEBUG_ECHOPGM_P" redefined
   92 |   #define DEBUG_ECHOPGM_P(P)          NOOP
      |
In file included from Marlin\src\gcode\bedlevel\ubl\../../../feature/bedlevel/ubl/ubl.h:29,
Compiling .pio\build\LPC1769\src\src\gcode\config\M309.cpp.o
                 from Marlin\src\gcode\bedlevel\ubl\../../../feature/bedlevel/bedlevel.h:67,
                 from Marlin\src\gcode\bedlevel\ubl\M421.cpp:32:
c:\users\light\onedrive\documents\github\marlin_q5\marlin\src\core\debug_out.h:62: note: this is the location of the previous definition
   62 |   #define DEBUG_ECHOPGM_P(P)      SERIAL_ECHOPGM_P(P)
      | 
In file included from Marlin\src\gcode\bedlevel\../../feature/bedlevel/ubl/ubl.h:29,
                 from Marlin\src\gcode\bedlevel\../../feature/bedlevel/bedlevel.h:67,
                 from Marlin\src\gcode\bedlevel\M420.cpp:28:
Marlin\src\gcode\bedlevel\../../feature/bedlevel/ubl/../../../core/debug_out.h:62: note: this is the location of the previous definition
   62 |   #define DEBUG_ECHOPGM_P(P)      SERIAL_ECHOPGM_P(P)
      |
In file included from Marlin\src\gcode\bedlevel\ubl\../../../feature/bedlevel/ubl/ubl.h:328,
                 from Marlin\src\gcode\bedlevel\ubl\../../../feature/bedlevel/bedlevel.h:67,
Compiling .pio\build\LPC1769\src\src\gcode\config\M43.cpp.o
                 from Marlin\src\gcode\bedlevel\ubl\G29.cpp:32:
c:\users\light\onedrive\documents\github\marlin_q5\marlin\src\core\debug_out.h:92: warning: "DEBUG_ECHOPGM_P" redefined
   92 |   #define DEBUG_ECHOPGM_P(P)          NOOP
Compiling .pio\build\LPC1769\src\src\gcode\config\M92.cpp.o
```

* Provide pictures or links to videos that clearly demonstrate the issue.
* See [Contributing to Marlin](https://github.com/MarlinFirmware/Marlin/blob/2.0.x/.github/contributing.md) for additional guidelines.

--- a/Marlin/src/core/debug_out.h
+++ b/Marlin/src/core/debug_out.h
@@ -27,7 +27,7 @@
 //
 
 #undef DEBUG_SECTION
-#undef DEBUG_PRINT_P
+#undef DEBUG_ECHOPGM_P
 #undef DEBUG_ECHO_START
 #undef DEBUG_ERROR_START
 #undef DEBUG_CHAR

